### DISCUSSION TITLE
Should TiddlyWiki default to read-only when served over http(s)?

### DISCUSSION THREAD

**Discussion by Jermolene** [TiddlyWiki/TiddlyWiki5#5558] [General]
Since the beginning, https://tiddlywiki.com/ has presented edit controls despite the fact that the wiki is read-only, and changes are not persisted. This was done to encourage visitors to look behind the scenes and experiment.

This has proved to be confusing for users in the context of tiddlywiki.com, but is potentially even more frustrating on hosting services like  TiddlyHost, TiddlySpot or Fission where users are encouraged to think that they're logged in and able to save changes even if they only have read-only access.

Meanwhile, TiddlyWiki in client-server configuration already engages read-only mode if the user cannot make changes, hiding the main editing controls.

So, I'm wondering if the time has come to switch the default so that TiddlyWiki's viewed on the web appear to be read-only unless specifically switched to read/write mode. That might happen if:

* The saver/syncer determines that the user is logged in and can save changes
* The user clicks an explicit button labelled "edit"

The wiki edit button might look like this:

![image](https://user-images.githubusercontent.com/174761/111911059-c4012b00-8a5b-11eb-82b3-59a26c41e1af.png)

**Comment by erickcampos50** [comment]
I think it would be a good idea. The looknfeel of a regular read-only  wiki would make visitors more comfortable.

**Reply by joshuafontany** [reply]
Agreed, and with the toggle available then 'power-users' can still edit directly on tiddlywiki.com a a test-bed, etc.

**Comment by twMat** [comment]
I like the general idea but I fear it may merely be an obstacle if it's not easy to *control* what shows and what doesn't: For public wikis there are often *many additional* things, beyond the edit controls, that one wants hidden or disabled. Unless such an "admin" type button can toggle *all* those things then it will just be in the way.

I see three general use cases that each would need their own tweakable default settings:

* Full transparency. *"Snoop around all you want, fellow *tiddleur*."*
* Semi-transparency. *"Yes, my fellow *tiddleur*, you *can* go backstage but you're not really supposed to in this TW.*" (Probably the case in the OP)
* Opaque. *"Is this Wordpress? Who cares; you're here for the information."*

**Reply by bmann** [reply]
I like this framing. The "is this wordpress?" is more likely to be static publishing as @abesamma points out. 

And: the more general, "which things do you want to hide?" would be pretty great to have in the toolkit.

**Comment by abesamma** [comment]
I think this is a good idea. I keep having to reinvent a pseudo read only
mode for my blog (abesamma.github.io) whenever I publish it in order to
avoid confusing people.

On the other hand, offering more options for publishing your tiddlers as
static pages for a headless CMS for example would negate the need for this,
but not 100% I think. You can still argue argue for it because it is at the
end so much simpler to just publish an entire TiddlyWiki than otherwise.
